src/coreclr/jit/importercalls.cpp

```diff
@@ -4410,28 +4399,25 @@ GenTree* Compiler::impSRCSUnsafeIntrinsic(NamedIntrinsic        intrinsic,
         }
 
         case NI_SRCS_UNSAFE_Write:
+        case NI_SRCS_UNSAFE_WriteUnaligned:
         {
             assert(sig->sigInst.methInstCount == 1);
 
             // ldarg.0
             // ldarg.1
+            // if NI_SRCS_UNSAFE_WriteUnaligned: unaligned. 0x01
             // stobj !!T
             // ret
 
-            return nullptr;
-        }
-
-        case NI_SRCS_UNSAFE_WriteUnaligned:
-        {
-            assert(sig->sigInst.methInstCount == 1);
+            GenTree* op1 = impPopStack().val;
+            GenTree* op2 = impPopStack().val;
 
-            // ldarg.0
-            // ldarg.1
-            // unaligned. 0x01
-            // stobj !!T
-            // ret
+            CORINFO_CLASS_HANDLE typeHnd = sig->sigInst.methInst[0];
+            ClassLayout*         layout  = nullptr;
+            var_types            type    = TypeHandleToVarType(typeHnd, &layout);
+            GenTreeFlags         flags   = intrinsic == NI_SRCS_UNSAFE_WriteUnaligned ? GTF_IND_UNALIGNED : GTF_EMPTY;
 
-            return nullptr;
+            return gtNewAssignNode(gtNewLoadValueNode(type, layout, op2, flags), op1);
```

Comment:
This needs to call `impAssignStruct` for `varTypeIsStruct(type)` values, otherwise it won't work correct for calls with return buffers and `MKREFANY`s.